### ISSUE TITLE
L4T R32.1 for Yocto/OE release Thud

### ISSUE THREAD

**Issue by leon-anavi** [madisongh/meta-tegra#152]
Hi @madisongh,

Are you interested in accepting a GitHub PR that backports L4T R32.1 with CUDA 10 from branch Warrior to a new branch compatible with Yocto/OE release Thud?

I have already done this and I interested in contributing it to the upstream. The benefit is that CUDA 10 requires GCC 7 and Poky for Yocto/OE release offers GCC 7 which allows using CUDA 10 without an external toolchain. If you are interested in this contribution could you please create a new branch with appropriate name, for example thud-l4t-r32.1, and I will make a GitHub pull request with the required changes for backporting?

Best regards,
Leon

**Comment by madisongh** [comment]
I thought I had run into some compatibility issues between OE-Core thud and some the L4T R32.1 binaries, but I don't recall now what those were.  If you have things working, great!  I've created a `thud-l4t-r32.1` branch as requested.

**Comment by leon-anavi** [comment]
Hi @madisongh,

I have created a GitHub Pull Request https://github.com/madisongh/meta-tegra/pull/153. I have copied what we have working in Warrior and adapted it for Thud therefore there are numerous changes. However I haven't added the machine definitions for Xavier and Nano yet. Probably it will work for them too but I don't have this hardware at the moment to test.

Using the current version from the GiHub PR I have built and successfully tested on NVIDIA Jetson TX2 core-image-minimal (based on Poky Thud) with cuda-samples, Linux kernel 4.9.140-l4t-r32.1+g3c02a65, GCC 7 and CUDA 10.

Best regards,
Leon
